ekf/tests/test_runner/result_check.c

```diff
@@ -0,0 +1,307 @@
+#include "../../logs/reader.h"
+
+#include <matrix.h>
+#include <hmap.h>
+#include <parser.h>
+
+#include <errno.h>
+#include <stdbool.h>
+#include <math.h>
+
+
+#define EPS_FIELDS_NB 5
+
+
+static struct {
+	matrix_t *expectedState;
+	matrix_t *eps;
+} ekfTestsResult_common;
+
+
+static int ekftests_expectedResultConverter(const hmap_t *h)
+{
+	int err = 0;
+	float eps;
+
+	err |= parser_fieldGetFloat(h, "attitude_a", &ekfTestsResult_common.expectedState->data[QA]);
+	err |= parser_fieldGetFloat(h, "attitude_i", &ekfTestsResult_common.expectedState->data[QB]);
+	err |= parser_fieldGetFloat(h, "attitude_j", &ekfTestsResult_common.expectedState->data[QC]);
+	err |= parser_fieldGetFloat(h, "attitude_k", &ekfTestsResult_common.expectedState->data[QD]);
+
+	err |= parser_fieldGetFloat(h, "gyro_bias_x", &ekfTestsResult_common.expectedState->data[BWX]);
+	err |= parser_fieldGetFloat(h, "gyro_bias_y", &ekfTestsResult_common.expectedState->data[BWY]);
+	err |= parser_fieldGetFloat(h, "gyro_bias_z", &ekfTestsResult_common.expectedState->data[BWZ]);
+
+	err |= parser_fieldGetFloat(h, "velocity_x", &ekfTestsResult_common.expectedState->data[VX]);
+	err |= parser_fieldGetFloat(h, "velocity_y", &ekfTestsResult_common.expectedState->data[VY]);
+	err |= parser_fieldGetFloat(h, "velocity_z", &ekfTestsResult_common.expectedState->data[VZ]);
+
+	err |= parser_fieldGetFloat(h, "accel_bias_x", &ekfTestsResult_common.expectedState->data[BAX]);
+	err |= parser_fieldGetFloat(h, "accel_bias_y", &ekfTestsResult_common.expectedState->data[BAY]);
+	err |= parser_fieldGetFloat(h, "accel_bias_z", &ekfTestsResult_common.expectedState->data[BAZ]);
+
+	err |= parser_fieldGetFloat(h, "position_x", &ekfTestsResult_common.expectedState->data[RX]);
+	err |= parser_fieldGetFloat(h, "position_y", &ekfTestsResult_common.expectedState->data[RY]);
+	err |= parser_fieldGetFloat(h, "position_z", &ekfTestsResult_common.expectedState->data[RZ]);
+
+	err |= parser_fieldGetFloat(h, "attitude_eps", &eps);
+	ekfTestsResult_common.eps->data[QA] = eps;
+	ekfTestsResult_common.eps->data[QB] = eps;
+	ekfTestsResult_common.eps->data[QC] = eps;
+	ekfTestsResult_common.eps->data[QD] = eps;
+
+	err |= parser_fieldGetFloat(h, "gyro_bias_eps", &eps);
+	ekfTestsResult_common.eps->data[BWX] = eps;
+	ekfTestsResult_common.eps->data[BWY] = eps;
+	ekfTestsResult_common.eps->data[BWZ] = eps;
+
+	err |= parser_fieldGetFloat(h, "velocity_eps", &eps);
+	ekfTestsResult_common.eps->data[VX] = eps;
+	ekfTestsResult_common.eps->data[VY] = eps;
+	ekfTestsResult_common.eps->data[VZ] = eps;
+
+	err |= parser_fieldGetFloat(h, "accel_bias_eps", &eps);
+	ekfTestsResult_common.eps->data[BAX] = eps;
+	ekfTestsResult_common.eps->data[BAY] = eps;
+	ekfTestsResult_common.eps->data[BAZ] = eps;
+
+	err |= parser_fieldGetFloat(h, "position_eps", &eps);
+	ekfTestsResult_common.eps->data[RX] = eps;
+	ekfTestsResult_common.eps->data[RY] = eps;
+	ekfTestsResult_common.eps->data[RZ] = eps;
+
+	return err;
```

Comment:
According to `parser_fieldGetFloat` description it returns _non-zero_ on error. Let's add proper check on return, as bitewise `or` summing may give strange results. It will definitely work, but it looks unsafe.
```suggestion
	return (err == 0) ? 0 : -1;
```